be/src/runtime/buffer_control_block.cpp

```diff
@@ -77,18 +77,13 @@ BufferControlBlock::BufferControlBlock(const TUniqueId& id, int buffer_size)
 
 BufferControlBlock::~BufferControlBlock() {
     cancel();
-
-    for (ResultQueue::iterator iter = _batch_queue.begin(); _batch_queue.end() != iter; ++iter) {
-        delete *iter;
-        *iter = NULL;
-    }
 }
 
 Status BufferControlBlock::init() {
     return Status::OK();
 }
 
-Status BufferControlBlock::add_batch(TFetchDataResult* result) {
+Status BufferControlBlock::add_batch(TFetchDataResultPtr& result) {
```

Comment:
Coding specification: All parameters passed by reference must be added with const, otherwise pointers are used

Reply:
can't use `const` here, the ownership will be transferred, i replaced it with value pass.